### DISCUSSION TITLE
Trace length analyzer vs tim_tot in stats csv

### DISCUSSION THREAD

**Discussion by Charlotte1001** [bsse-scf/pyMINFLUX#10] [Q&A] [answered]
Is time_tot in ms that one can calculate from the original column 'tim' in the raw data and the trace length plotted in your analyzer the same? The trace length in the plotted analyzer provide a shorter medium trace length in my data set then when I calculate the trace length from tim_tot. Thanks for your help in advance!

**Comment by aarpon** [comment, marked as answer]
This is how the various tracking stats are currently calculated:

* `tim_tot` is the sum of all time intervals between consecutive localizations of a trace **scaled by 1,000** with respect to the raw numbers in the original NumPy array to have measures **in ms** in the stats: it's basically `1e3 * sum(diff(tim))` for each `tid`;
* the `trace_length` is the sum of all `x, y, z` displacements along each trace (steps are calculated as the Euclidean distance between `pos(t-1)` and `pos(t)`);
* the average trace speed in `trace_length` / `tim_tot`.
